tasks/task_wifi.c

```diff
@@ -36,23 +52,64 @@ static void task_wifi_scan_handler(retro_task_t *task)
    task_set_finished(task, true);
 }
 
-bool task_push_wifi_scan(retro_task_callback_t cb)
+static void task_wifi_enable_handler(retro_task_t *task)
```

Comment:
In all these task functions, you need to return early if `task` is `NULL`. This is also true of `task->user_data`, where used.

The existing code was sloppy in this, so I fully understand why you didn't add these checks :)